lighthouse-core/gather/gatherers/inspector-issues.js

```diff
@@ -48,18 +58,25 @@ class InspectorIssues extends Gatherer {
     driver.off('Audits.issueAdded', this._onIssueAdded);
     await driver.sendCommand('Audits.disable');
     const artifact = {
-      /** @type {Array<LH.Crdp.Audits.MixedContentIssueDetails>} */
+      blockedByResponse: [],
       mixedContent: [],
     };
 
     for (const issue of this._issues) {
-      if (issue.details.mixedContentIssueDetails) {
-        const issueDetails = issue.details.mixedContentIssueDetails;
-        const issueReqId = issueDetails.request && issueDetails.request.requestId;
-        if (issueReqId &&
-          networkRecords.find(req => req.requestId === issueReqId)) {
-          artifact.mixedContent.push(issue.details.mixedContentIssueDetails);
-        }
+      const details = getDetails(issue);
+      if (!details) continue;
+
+      const issueRequestId = details.request && details.request.requestId;
+      const issueRequest = networkRecords.find(req => req.requestId === issueRequestId);
+
+      // Expected a request, but didn't find it.
+      if (issueRequestId && !issueRequest) continue;
+
+      const artifactKey = /** @type {keyof typeof artifact} */ (
```

Comment:
there's no type safety here, so I'm not sure it's worth pretending there is :)

If we're really going to be using all the Issues, maybe we should drop our custom format in favor of just keeping the artifact as the protocol provides: `Array<LH.Crdp.Audits.InspectorIssue>`.

We could even upstream a change for discriminated unions without breaking the current type, if `InspectorIssue` was
```ts
export type InspectorIssue = SameSiteCookieIssue | MixedContentIssue | BlockedByResponseIssue;
export interface SameSiteCookieIssue {
  code: `SameSiteCookieIssue`,
  details: {
    sameSiteCookieIssueDetails: SameSiteCookieIssueDetails;
  };
};
export interface MixedContentIssue {
  code: `MixedContentIssue`,
  details: {
    mixedContentIssueDetails: MixedContentIssueDetails;
  };
};
export interface BlockedByResponseIssue {
  code: `BlockedByResponseIssue`,
  details: {
    blockedByResponseIssueDetails: BlockedByResponseIssueDetails;
  };
};
```
any existing code could use it with no changes, but now tsc could reason about it.

(it is still `experimental`, so if they want to change it to drop the nested `details`, all the better :)

Reply:
> there's no type safety here, so I'm not sure it's worth pretending there is :)

Would you prefer a tsignore over everything?

> drop our custom format

```
for (const issue of artifacts.InspectorIssues) {
	if (!issue.details.mixedContentIssueDetails) continue;

    // do something ...
	issue.details.mixedContentIssueDetails;
}
// vs
for (const issue of artifacts.InspectorIssues.mixedContent) {
    // do something ...
	issue.details.mixedContentIssueDetails;
}
```

I don't care to argue for or against this. Clearly the latter is preferable, and imo the massaging done in the gatherer to enable it is of minimal complexity. Granted, the ergonomics savings is small.

> (it is still experimental, so if they want to change it to drop the nested details, all the better :)

Is it a matter of one of us making this change or them? I don't think it's worth our time tbh but if you happen to be familiar with how the TSC is generated maybe it won't take long.